core/consensus/aura/src/lib.rs

```diff
@@ -464,11 +506,13 @@ impl<B: Block, C, E, MakeInherent, Inherent> Verifier<B> for AuraVerifier<C, E,
 		justification: Option<Justification>,
 		mut body: Option<Vec<B::Extrinsic>>,
 	) -> Result<(ImportBlock<B>, Option<Vec<Ed25519AuthorityId>>), String> {
-		use runtime_primitives::CheckInherentError;
 		const MAX_TIMESTAMP_DRIFT_SECS: u64 = 60;
 
-		let (timestamp_now, slot_now) = timestamp_and_slot_now(self.slot_duration.0)
-			.ok_or("System time is before UnixTime?".to_owned())?;
+		let mut inherent_data =
+			self.inherent_data_providers.create_inherent_data().map_err(String::from)?;
+		let (timestamp_now, slot_now) =
+			extract_timestamp_and_slot(&inherent_data)
```

Comment:
just to make sure I understand this, the above call to `create_inherent_data` generates the current timestamp and slot, and then below, we only overwrite the `slot_num` so that verification fails if there's a mismatch?

Reply:
Yeah that is done.
I did not change any logic, I just ported it to the new implementation.
As I understood it, the `current_slot_num` is generated by `InherentData` and then we replace this current slot with the slot, sealed in the checked header.